posts/inside-rust/2020-11-02-Using-rustc_cranelift_codegen.md

```diff
@@ -0,0 +1,148 @@
+---
+layout: post
+title: "Using rustc_codegen_cranelift for debug builds"
+author: Joshua Nelson
+team: The Compiler Team <https://www.rust-lang.org/governance/teams/compiler>
+---
+
+## What is `rustc_codegen_cranelift`?
+
+[`rustc_codegen_cranelift`], or just `cg_clif` for short, is a new experimental
+codegen backend for the Rust compiler. The existing backend is LLVM, which very
+good at producing fast, highly optimized code, but is not very good at
+compiling code quickly. `cg_clif`, which uses the [Cranelift] project, would
+provide a fast backend which greatly improves compile times, at the cost of
+only performing very few optimizations. This is a great fit for debug builds, and the hope is
+that `cg_clif` will eventually be the default backend in debug mode.
+
+## What is the progress of using `rustc_codegen_cranelift` for debug builds?
+
+There has been an [Major Change Proposal][MCP] open for some time for making
+`cg_clif` part of the main Rust repository. Recently, [the MCP was
+accepted][compiler-team#270] and the compiler team [merged][#77975]
+`rustc_cranelift_codegen` [into the main Rust git repository][#77975].
+`cg_clif` still is not distributed with `rustup`, but this means you can now
+build it from source in-tree!
+
+## How do I use `rustc_codegen_cranelift`?
+
+In this section, I'll walk through step-by-step how to build the new backend from source, then use it on your own projects. All code is copy/paste-able, and each step is explained.
+
+First, let's build `cg_clif` from source.
+
+```sh
+$ git clone https://github.com/bjorn3/rustc_codegen_cranelift.git
+$ ./prepare.sh
+$ cargo build --release
+$ ./build_sysroot/build_sysroot.sh
+```
+
+Now, we can start using it for our own projects. For demonstration
```

Comment:
```suggestion
Now, we can start using it in a project. For demonstration
```

Reply:
'in a project' sounds like it's a dependency. Maybe 'to compile a project'?

Reply:
Sure, something like that sounds fine. The "our own" part is what sounds funny, because it implies that you can use it for some projects beforehand, but not personal projects.